### DISCUSSION TITLE
Edge runtime and OpenTelemetry

### DISCUSSION THREAD

**Discussion by bguthrie** [vercel/community#2253] [Help] [answered]
### Summary

We rely heavily on OpenTelemetry for tracking function metrics, but I can't even require the OpenTelemetry types from their Node libs in an edge runtime without bumping into incompatibility issues—it seems that OTel is coupled directly to Node. (E.g., I can't `import { Span } from `@opentelemetry/api`.) Has anyone else run into this issue? Any known workarounds in Vercel?

**Comment by amyegan** [comment, marked as answer]
Hi @bguthrie! The [Vercel OpenTelemetry Collector](https://vercel.com/docs/concepts/observability/otel-overview/quickstart#) is in [beta](https://vercel.com/docs/concepts/release-phases#beta) so there are some current limitations and things may change as development progresses. At the moment, the OTEL collector can only be used in a Serverless Function on a project that uses an OTEL Integration.

**Comment by bguthrie** [comment]
Hi Amy! Thanks for the response. We're on a Pro plan and I *believe* we're
using the Vercel OTel collector, but my problem isn't at runtime—my NextJS
instance in localhost errors out simply loading ***@***.***/api`. Since
the official docs reference, this might be a version issue—I'll follow up
and let you know.
